src/core/providers/arcgis/qgsarcgisrestutils.cpp

```diff
@@ -748,6 +747,84 @@ std::unique_ptr<QgsMarkerSymbol> QgsArcGisRestUtils::parseEsriPictureMarkerSymbo
   return symbol;
 }
 
+std::unique_ptr<QgsMarkerSymbol> QgsArcGisRestUtils::parseEsriTextMarkerSymbolJson( const QVariantMap &symbolData )
+{
+  QgsSymbolLayerList layers;
+
+  QString fontFamily;
+  fontFamily = symbolData.value( QStringLiteral( "font" ) ).toMap().value( QStringLiteral( "family" ) ).toString();
+
+  QString chr;
+  chr = symbolData.value( QStringLiteral( "text" ) ).toString();
+
+  double pointSize;
+  pointSize = symbolData.value( QStringLiteral( "font" ) ).toMap().value( QStringLiteral( "size" ) ).toDouble();
+
+  QColor color = convertColor( symbolData.value( QStringLiteral( "color" ) ) );
+
+  double angle;
+  angle = symbolData.value( QStringLiteral( "angle" ) ).toDouble();
```

Comment:
It's tricky to confirm without a server to check against, but I suspect we need to transform the rotation here. From the ESRI docs: "The rotation is from East in a counter-clockwise direction where East is the 0° axis.", whereas QGIS symbol markers use clockwise rotation from north. So we'll need to do a "90-angle from JSON" modification here.